genericapi/handler.go

```diff
@@ -84,15 +92,28 @@ func (h *Handler) ServeCreateAlert(w http.ResponseWriter, r *http.Request) {
 
 	summary := r.FormValue("summary")
 	details := r.FormValue("details")
+	action := r.FormValue("action")
 
-	summary = validate.SanitizeText(summary, alert.MaxSummaryLength)
-	details = validate.SanitizeText(details, alert.MaxDetailsLength)
+	contentType := r.Header.Get("Content-type")
+	if contentType == "application/json" {
+		requestBoday, _ := ioutil.ReadAll(r.Body)
+
+		var alertMessage AlertMessage
```

Comment:
For this, let's do an inline struct since we don't need it elsewhere. Also we can ignore the `json` tags since Go parses JSON ignoring case already. Lastly, let's make the fields pointers so we can fallback to query parameters for any missing fields (this follows the current behavior of `FormValue`)

```suggestion
		var b struct {
			Summary, Details, Action *string
		}
```